# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsRefreshDriver.cpp
# Commit: bcfcd8b57192
# Full Hash: bcfcd8b5719248860bc26a6fce935b516ecb042e
# Author: Daniel Holbert <dholbert@cs.stanford.edu>
# Date: 2023-10-10 09:23:29
# Description:
#   Bug 1857561: Make nsRefreshDriver::Tick gracefully bail if a resize observer destroys the PresContext. r=surkov,emilio
#   
#   Note, I'm deleting the null-check from UpdateIntersectionObservations, since
#   I'm essentially hoisting it up one level (and I don't think there's anything
#   that could null out mPresContext in the intervening code).
# ==============================================================================

diff -r edb416a08b6e -r bcfcd8b57192 layout/base/nsRefreshDriver.cpp
--- a/layout/base/nsRefreshDriver.cpp	Mon Oct 09 18:20:23 2023 +0000
+++ b/layout/base/nsRefreshDriver.cpp	Mon Oct 09 18:49:39 2023 +0000
@@ -2213,10 +2213,6 @@
 void nsRefreshDriver::UpdateIntersectionObservations(TimeStamp aNowTime) {
   AUTO_PROFILER_LABEL_RELEVANT_FOR_JS("Compute intersections", LAYOUT);
 
-  if (MOZ_UNLIKELY(!mPresContext)) {
-    return;
-  }
-
   AutoTArray<RefPtr<Document>, 32> documents;
 
   if (mPresContext->Document()->HasIntersectionObservers()) {
@@ -2749,9 +2745,14 @@
     pm->UpdatePopupPositions(this);
   }
 
-  // Notify resize obsrevers if any, see
+  // Notify resize observers if any, see
   // https://html.spec.whatwg.org/#update-the-rendering step 14.
   NotifyResizeObservers();
+  if (MOZ_UNLIKELY(!mPresContext || !mPresContext->GetPresShell())) {
+    // A resize observer callback apparently destroyed our PresContext.
+    StopTimer();
+    return;
+  }
 
   // Update the relevancy of the content of any `content-visibility: auto`
   // elements. The specification says: "Specifically, such changes will
